# Dorfman-Feature-selection
We introduce the Dorfman variable screening framework, a novel, computationally efficient Variable Selection method for high-dimensional, group correlated variables. It is a multi-stage hierarchical procedure that adapts group testing principles to high-dimensional regression. The procedure first clusters genes into data-driven, biologically meaningful groups using hierarchical clustering based on Pearson correlation or sparse correlation estimated via the graphical LASSO, which induces sparsity in the inverse covariance matrix and stabilizes clustering in high dimensions (Friedman et al., 2008). Group-level global tests are then performed to identify candidate signal groups, followed by within-group individual variable screening for identified groups. A final regularized selection step using the elastic net or LASSO refines variable selection while accounting for residual correlation structure (Tibshirani, 1996; Zou & Hastie, 2005). Robust variants of the Dorfman framework further address outlier contamination by incorporating Huber-weighted regression throughout group testing and within-group screening (Huber, 1964), with groups determined by dynamic cut hierarchical clustering (Langfelder et al., 2008) using Spearman correlation or sparse correlation derived from an OGK (orthogonalized Gnanadesikanâ€“Kettenring) covariance estimator (Oellerer & Croux, 2015). Together, in this modular structure, these methods provide a unified, scalable, and robust approach to screening grouped predictors in ultra-high-dimensional biomedical data, closely aligned with modern developments in group testing and Bayesian extensions accounting for noise and dilution effects (Tatsuoka, Chen & Lu, 2022).  Focus here will be given to continuous outcomes in linear regression models, including when deviations from classical assumptions arise. Simulations are conducted that show the advantages of our proposed Dorfman-based framework, particularly in non-normal data settings. Finally, a real-data example is presented that illustrates how the methods are used to identify genes that predict IC50 for a MEK-inhibitor in non-small cell lung cancer with bulk RNA sequencing data.<img width="468" height="418" alt="image" src="https://github.com/user-attachments/assets/6bd051b8-a06a-4288-9ceb-cbbeefc2dacb" />

