# Dorfman-Feature-selection
We introduce the Dorfman variable screening framework, a novel, computationally efficient Variable Selection method for high-dimensional, group correlated variables. It is a multi-stage hierarchical procedure that adapts group testing principles to high-dimensional regression. The procedure first clusters genes into data-driven, biologically meaningful groups using hierarchical clustering based on Pearson correlation or sparse correlation estimated via the graphical LASSO, which induces sparsity in the inverse covariance matrix and stabilizes clustering in high dimensions (Friedman et al., 2008). Group-level global tests are then performed to identify candidate signal groups, followed by within-group individual variable screening for identified groups. A final regularized selection step using the elastic net/ adaptive EN refines variable selection while accounting for residual correlation structure (Tibshirani, 1996; Zou & Hastie, 2005). Robust variants of the Dorfman framework further address outlier contamination by incorporating Huber-weighted regression throughout group testing and within-group screening (Huber, 1964), with groups determined by dynamic cut hierarchical clustering (Langfelder et al., 2008) using Spearman correlation or sparse correlation derived from an OGK (orthogonalized Gnanadesikanâ€“Kettenring) covariance estimator (Oellerer & Croux, 2015). Together, in this modular structure, these methods provide a unified, scalable, and robust approach to screening grouped predictors in ultra-high-dimensional biomedical data, closely aligned with modern developments in group testing and Bayesian extensions accounting for noise and dilution effects (Tatsuoka, Chen & Lu, 2022). Focus here will be given to continuous outcomes in linear regression models, including when deviations from classical assumptions arise. Simulations are conducted that show the advantages of our proposed Dorfman-based framework, particularly in non-normal data settings. Finally, a real-data example is presented that illustrates how the methods are used to identify genes that predict IC50 for a MEK-inhibitor in non-small cell lung cancer with bulk RNA sequencing data. 

We demonstrate the advantage of our Dorfman methods in two simulation scenarios for a continous outcome, 1) Normal data satisfying LR assumptions, n=200, p=500, 100 groups, 20 groups with each correlation (0.1, 0.3, 0.5, 0.7, 0.9), corresponding to standardized beta values of (1, 0.9, 0.7, 0.5, 0.3). 2) Corrrupted data with the same simulated correlation except standardized beta values of (2.0, 1.8, 1.5, 1.2, 1.0), along with X corrupted by: Broken within-group correlations, Fake cross-group correlations, Batch effects (large mean shifts), Extreme leverage outliers and Asymmetric contamination (skewed distortions), and Y with Heavy-tailed noise, Heteroscedastic variance and Asymmetric noise shifts. In both cases, there are 5 true groups with 5 variables distributed within each of the 20 groups respectively. Our results show that Dorfman-Sparse-adaptive EN using the graphical LASSO achieved superb performance under normal scenario (F1 = XXX, RMSE = XXX), followed by robust-Dorfman-OGK-adaptive-EN using the OGK cov matrix as well as incorporating Huber weights (F1 = XXX, RMSE = XXXX). Interestingly, in the robust case, we found that robust-Dorfman-OGK-adaptiveEN as well as robust dorfman-OGK-EN performed best (F1 = XXX, and XXX; RMSE = XXX and XXX), significantly outperforming classical Dorfman-Sparse-adaptive EN. In both cases, dorfman methods outperform competitors including adaptive EN, EN, SIS-LASSO, and gAR2 (the best performing method under this simulation scenario by Qiu & Ahn, 2019). The plots are shown below: 


Applied to GDSC RNA-seq data for trametinib (MEK inhibitor) response prediction in non-small cell lung cancer, Dorfman methods achieved superior predictive accuracy (RMSE=2.17-2.43) compared to adaptive LASSO/EN (RMSE=2.41-2.46) and traditional approaches SIS-LASSO, LASSO and EN (RMSE=2.49-2.84), with robust dorfman methods achieving superb performance, echoing the fact that data is non-normal, with outliers and violations LR assumptions. When signature genes for trametinib response in NSCLC are grouped into tiers according to literature evidence, robust Dorfman selections showed significant enrichment for genes of tiers 1-2 (clinical hits) compared to adaptive LASSO/EN, SIS-LASSO and vanilla EN/LASSO, with robust Dorfman-EN achieving the highest clinical ratio. Interestingly Moreover, genes exclusively uncovered by robust dorfman (not classical) exhibited higher deviation from normality (JB score) along with extreme kurtosis and skewness.

<img width="215" height="118" alt="image" src="https://github.com/user-attachments/assets/59ba3895-538b-4f47-a7f0-593d4702814e" />

<img width="182" height="113" alt="image" src="https://github.com/user-attachments/assets/3736eb8e-d3d0-4dd0-9c2d-eca02ffd769f" />

Our results demonstrate that robust-OGK-Dorfman-Adaptive EN/ EN as the method of choice for feature selection of high-dimensional, group correlated data. We plan to extend this to ultra-high dimension (e.g. p=100,000), incorporating improvements in the first step such as block OGK, block graphical lasso, and LSH in group merging as well as bayesian optimized hyperparameter selection instead of CV, with use of HPC clusters. 







